packages/appstore/src/components/CFDs/cfd-demo-accounts.tsx

```diff
@@ -0,0 +1,181 @@
+            ),
+            disabled: false,
+            platform: CFD_PLATFORMS.DXTRADE,
+            type: 'synthetic',
+            // ToDo: deriv x should have type of all in new API
+            // type: 'all'
+        },
+    ];
+
+    const { client, modules, common, ui }: TRootStore = useStores();
+    const {
+        standpoint,
+        dxtrade_tokens,
+        createCFDAccount,
+        setCurrentAccount,
+        setMT5TradeAccount,
+        toggleMT5TradeModal,
+        enableCFDPasswordModal,
+        openAccountNeededModal,
+        has_maltainvest_account,
+    } = modules.cfd;
+    const { platform, setAppstorePlatform } = common;
+    const { openTopUpModal } = ui;
+    const { is_eu } = client;
+
+    const openAccountTransfer = (
+        data: DetailsOfEachMT5Loginid & { account_id?: string; platform?: string },
+        meta: { category: string; type?: string }
+    ) => {
+        setCurrentAccount(data, meta);
+        openTopUpModal();
+    };
+
+    const REAL_DXTRADE_URL = 'https://dx.deriv.com';
+    const DEMO_DXTRADE_URL = 'https://dx-demo.deriv.com';
+
+    const getDXTradeWebTerminalLink = (category: string, token?: string) => {
+        let url = category === 'real' ? REAL_DXTRADE_URL : DEMO_DXTRADE_URL;
+
+        if (token) {
+            url += `?token=${token}`;
+        }
+
+        return url;
+    };
+
+    const openCFDAccount = (account_type: string) => {
+        if (is_eu && !has_maltainvest_account && standpoint.iom) {
+            openAccountNeededModal('maltainvest', localize('Deriv Multipliers'), localize('demo CFDs'));
+        } else {
+            createCFDAccount({
+                category: 'demo',
+                type: account_type,
+                platform,
+            });
+            enableCFDPasswordModal();
+        }
+    };
+
+    const existingDemoAccounts = (existing_platform: TPlatform, market_type?: string) => {
+        const acc = Object.keys(current_list).some(key => key.startsWith(`${existing_platform}.demo.${market_type}`))
+            ? Object.keys(current_list)
+                  .filter(key => key.startsWith(`${existing_platform}.demo.${market_type}`))
+                  .reduce((_acc, cur) => {
+                      _acc.push(current_list[cur]);
+                      return _acc;
+                  }, [] as TDetailsOfEachMT5Loginid[])
+            : undefined;
+        return acc;
+    };
+
+    return (
+        <div className='cfd-demo-account'>
+            <div className='cfd-demo-account__accounts'>
+                {available_demo_accounts.map(
+                    account =>
+                        account.is_visible && (
+                            <div className={`cfd-demo-account__accounts--item ${account.name}`} key={account.name}>
+                                {existingDemoAccounts(account.platform, account.type) ? (
+                                    existingDemoAccounts(account.platform, account.type)?.map(existing_account => {
```

Comment:
```suggestion
                                    existing_demo_accounts.map(existing_account => {
```
No need for optional chaining as the check has already been done before this statement